cabal-install/src/Distribution/Client/IndexUtils/IndexState.hs

```diff
@@ -123,22 +123,22 @@ insertIndexState rn idx (TIS def m)
 -------------------------------------------------------------------------------
 
 -- | Specification of the state of a specific repo package index
-data RepoIndexState
+data IndexState
```

Comment:
I'm not sure I quite like this renaming: are you sure it improves things? I think not because:

 1. `RepoIndexState` expresses more important information than `IndexState` (there may be more indecies in the project than repository index, potenatially)
 2. The diff grows significantly, it seems.

Any pros? I am not an expert in the actual index-related code, so maybe this change better reflects semantics. If you think so, I'm happy to follow.

Reply:
IMHO The definition of `IndexState` seems to be very simple and generic (it answers: "what can an index-state be"?). It is then used in `TotalIndexState` to associate an `IndexState` to each `Repository`.

> there may be more indecies in the project than repository index, potenatially

:thinking: that's fair, indeed I didn't think of that. Notice we do already use `index-state:` in cabal.project, which seems to preclude a more general meaning of "index state".

> The diff grows significantly, it seems.

I am happy to move this to a separate PR. I admit it was a bit cheeky of me to include it here :relaxed:

Reply:
I'm for omitting the rename or moving it to a separate pr. The name isn't too long, and it's another breaking change

Reply:
:+1: